### PR TITLE
Assignment4 otp

### PR DESCRIPTION
**Pull request by shashitekula**
Hi Professor,

Please find the update code base for the assignment  4 OTP.

My name : Shashi Tekula
email : stekula@smu.edu
ID : 46739619

Thank you
Shashi Tekula

### CODE REVIEW

**Comment by shashitekula** [comment]
Hi professor,

FYI ...
in the local environment , was not able to find file resource , so i have mapped it to absolute path to make it work
 @word_list_file_name "/Users/stekula/smu/assignment4-otp-hangman/assets/words.8800"

File: lib/hangman.ex [hunk 1]
```diff
@@ -1,23 +1,26 @@
 defmodule Hangman do
-  use Application
+  use Application,Supervisor
 
   @moduledoc """
 
-  Write your description of your supervision scheme here...
+  @Strategies - Hangman supervisor defines "one_for_one"" strategies to manage child processes.
```
Comment (pragdave) [inline lib/hangman.ex:6]: # 7

I didn't find this very clear.

Also, you might want to consider rest_for_one at the top level.

File: lib/hangman/dictionary.ex [hunk 2]
```diff
@@ -1,12 +1,33 @@
 defmodule Hangman.Dictionary do
+    use GenServer
+
+
+    ###################
+    # External API
+
+    def start_link(default) do
```
Comment (pragdave) [inline lib/hangman/dictionary.ex:8]: # -3

I was hoping you'd put the word_list into the state (in `init`). That way the various `handle_xxx` functions wouldn't reread it each time.

File: lib/hangman/game_server.ex [hunk 3]
```diff
@@ -0,0 +1,77 @@
+defmodule Hangman.GameServer do
+  @moduledoc false
+  use GenServer
+  def start_link(current_word\\Hangman.Dictionary.random_word) do
```
Comment (pragdave) [inline lib/hangman/game_server.ex:4]: # -3

I'd really like to see the use of a module tag (like `@me`) to remove the duplication of the server process name throughout.

**Pull request closed without merging**